### PR TITLE
Add livers and stomachs

### PR DESCRIPTION
**Pull request by Incoming5643**
Without a stomach:
- You can't puke to reduce your toxins
- You can't eat or drink
- You can't devour anything

Without a liver:
- You can't get sober, ever (unless you take antihol)

Xenos get xeno versions of the stomach and liver, the xeno stomach allows for devouring of all carbons and the liver removes any drunkenness (and likewise a xeno with a removed or replaced liver can now get drunk).

Like all xeno organs these will function in humans if transplanted, but humans have to be fat to devour even with the xeno stomach.

Also adds a robotic version of the liver simply because the sprite was there, but doesn't meaningfully implement it.

If anyone has some ideas for more functionality/special mutantrace versions feel free to suggest em.

!!pull has not been tested yet!!

### CODE REVIEW

**Comment by Zilenan91** [comment]
How about a lizard one that lets them eat cockroaches

**Comment by Incoming5643** [comment]
There are also preexisting sprites for kidneys I just can't really think of what functions to assign to them since we don't simulate spessmen bladders.

**Comment by Zilenan91** [comment]
Kidneys filter toxins/bad shit in your food, so maybe multiply toxin damage in someone if they don't have kidneys.

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -378,6 +369,54 @@
...
+		if(drunkenness >= 81)
+			adjustToxLoss(0.2)
+			if(prob(5) && !stat)
+				src << "<span class='warning'>Maybe you should lie down for a bit...</span>"
+
+		if(drunkenness >= 91)
+			adjustBrainLoss(0.4)
+			if(prob(20) && !stat)
+				if(SSshuttle.emergency.mode == SHUTTLE_DOCKED && z == ZLEVEL_STATION) //QoL mainly
+					src << "<span class='warning'>You're so tired... but you can't miss that shuttle...</span>"
```
Comment (Cheridan) [inline code/modules/mob/living/carbon/life.dm]: still lame

Comment (Incoming5643) [inline code/modules/mob/living/carbon/life.dm]: I agree 100% and would remove it if it was the scope of the PR

**Comment by coiax** [comment]
The internet suggests that a lack of a liver is fatal in the long term, but is short on details.

My layman biology suggests that slowly various toxins will build up in the bloodstream that the liver would normally filter out, which end up killing you at high levels. Not too difficult to simulate, actually, probably just make the absence of a liver build up some sort of harmful reagent which at high levels starts doing increasing amounts of toxic and brute damage, maybe start damaging the other organs.

But, coiax you say, couldn't you just use a purgative chemical to remove this harmful chemical? Yes, that's the point, you're replacing the liver's function with chemistry to let them live longer.

To be honest, I think the kidneys do something similar (kidney failure seems a lot more common than liver failure).

**Comment by PKPenguin321** [comment]
lack of a liver should make deadly chemicals purge slower

**Comment by coiax** [comment]
well, we might be hitting up at the edge of the abstraction wall with our reagent based human body. Do Calomel, Charcoal and Pentacid work through the liver? Or by themselves?

**Comment by Iamgoofball** [comment]
by themselves

keep reagent code in reagent code

what we CAN do however is make the actual "remove the chems" aspect of metabolism be effected by whether you have a liver or not

slice it in half or some shit idfk

**Comment by PKPenguin321** [comment]
>what we CAN do however is make the actual "remove the chems" aspect of metabolism be effected by whether you have a liver or not

see this is what i was thinking but then
>take 5u of every healing chem
>infinite healing because chem never metabolizes

so why not instead make only harmful chems not metabolize so as to follow the *toxins* purging theme? idk

**Comment by Core0verload** [comment]
I like it. Anything that makes organs more useful is :+1: from me.

File: code/modules/surgery/organs/organ_internal.dm [hunk 2]
```diff
@@ -246,6 +244,70 @@
...
+			owner.nutrition += 10
+
+/obj/item/organ/internal/stomach/proc/check_devourabilty(mob/living/carbon/C)
+	if(istype(C, can_devour))
+		return 1
+	return 0
+
+/obj/item/organ/internal/stomach/alien
+	name = "alien stomach"
+	desc = "An unnevingly huge meat sack for holding unnerved meatsacks."
```
Comment (RemieRichards) [inline code/modules/surgery/organs/organ_internal.dm]: unnervingly

**Comment by WJohn** [comment]
We need to be able to eat these, or at least turn them into dishes.

**Comment by FantasticFwoosh** [comment]
Have you considered transplanting a liver from a polar bear/space bear via butchering?

To my loose knowledge they have extremely efficient livers (a polar bear liver can kill you with vitamin A overdose) because removing the drunkenness state might be boring in the bar long term, but im sure people would enjoy drinking all that they want, getting a bit tipsy but not puking or becoming severely ill with it wearing off quickly would be a more enjoyable experience via  a second tier and less dangerous transplant..

Otherwise really nice PR work.

**Comment by Shapsy** [comment]
@PKPenguin321 how would you classify what is/isn't harmful

**Comment by RemieRichards** [comment]
@Shapsy a good start might be the ```/toxin``` path.

**Comment by Core0verload** [comment]
@WJohn, you actually can eat organs, just like any other food item.

**Comment by RemieRichards** [comment]
Except for brains, because that's bad for balance.

**Comment by coiax** [comment]
RIP brain burgers

**Comment by RemieRichards** [comment]
I didn't really mean at all, I mean in regards to how organs work, with an organ you just pick it up and shove it in your mouth and it gets silently converted to a food object, that's not done with a brain as it would instantly kill (well, you know) the player who's brain it was. with proper cooking it's atleast not as simple as "pick it up and use it" to keep a player out.

**Comment by Zilenan91** [comment]
Oh also btw having no liver should make you not able to process nutriment. Livers in real life are the organ that stores/uses the carbohydrates that you eat in your food.

**Comment by lordpidey** [comment]
How about no liver means all overdose thresholds are halved?

**Comment by coiax** [comment]
@RemieRichards I suppose the thing is, if you have a player's brain, statistically they tend to be dead anyway.

NEW FEATURE, eating someone's brain has a change of absorbing some of their knowledge (wooooo spoooky), and an even smaller chance of their personality being permanently added to yours so they can bitterly complain about how shit you are

**Comment by Ultimate-Chimera** [comment]
Odd, but nice. :+1:

**Comment by coiax** [comment]
@Incoming5643 Reminder that all mob/living/carbons have stomach_contents, although it's barely used apart from xeno swallowing (and then regurgitating), and changeling eggs bursting out of corpses stomaches.

**Comment by coiax** [comment]
Oh, and fat humans can [swallow monkeys whole](https://github.com/tgstation/-tg-station/blob/e3a337efd6f6a2ed384e5efa2a0e487231258377/code/modules/mob/mob_grab.dm#L204).

**Comment by Core0verload** [comment]
Em. How about TESTING and MERGING it already?

**Comment by Incoming5643** [comment]
There's more I want to do with it. I've just been pretty busy with LIFE stuff.

**Comment by Core0verload** [comment]
When life gets in a way, just commit what you have and finish it later.

**Comment by Incoming5643** [comment]
I'll finish this soonish. Sometime before next Wednesday at the latest, as things might get **intense** after that.

**Comment by AlexanderUlanH** [comment]
Why not turn REM from a define into a liver variable? In most chems, it multiplies both the reagent consumption rate and the effect per tick of the chemical.

**Comment by Incoming5643** [comment]
Lots of great ideas here I want to impliment, but all drive to finish it has died for reasons I cannot properly explain. Will return to this eventually, but for the time being this might just end up holding back better stuff like dismemberment.

**Comment by Core0verload** [comment]
Maybe I'll do a takeover on your PR.

**Pull request closed without merging**